stl/inc/utility

```diff
@@ -927,6 +927,41 @@ _EXPORT_STD [[noreturn]] __forceinline void unreachable() noexcept /* strengthen
 #endif // defined(_DEBUG)
 }
 
+#ifdef __clang__
+// Clang 17 implements forward_like directly but is confused by our deduced return type

```

Comment:
I recommend filing a Clang bug; our current implementation is indeed clearer thanks to `if constexpr` locality.

Reply:
I've re-opened LLVM-64029.